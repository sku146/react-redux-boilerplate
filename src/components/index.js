// @flow
import LandingPageComponent from './module/landing-page';

export {
  LandingPageComponent,
};

export default {};
