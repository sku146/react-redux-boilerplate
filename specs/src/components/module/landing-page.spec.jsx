import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { LandingPageComponent } from 'components';

describe('Landing Page Component', () => {
  let defaultProps = {
    actions: {
      application: {
        applicationStarted: jest.fn()
      }
    },
    apiActions: {
      exampleList: {
        fetchExampleList: jest.fn(),
      },
    },
    exampleList: ['test', 'test1', 'test2'],
  };

  const ShallowComponent = (props) => {
    return shallow(<LandingPageComponent {...props} />);
  };

  const RendererComponent = (props) => {
    return renderer.create(<LandingPageComponent {...props} />);
  };

  describe('render', () => {
    it('component should render as expected', ()=> {
      const component = RendererComponent(defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

  describe('componentDidMount', () => {
    it('should call applicationStarted action', () => {
      const component = ShallowComponent(defaultProps);
      component.instance().componentDidMount();
      expect(defaultProps.actions.application.applicationStarted).toHaveBeenCalled();
      expect(defaultProps.apiActions.exampleList.fetchExampleList).toHaveBeenCalled();
    });
  });
});
