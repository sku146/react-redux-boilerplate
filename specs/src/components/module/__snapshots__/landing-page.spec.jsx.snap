// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Component render component should render as expected 1`] = `
<div
  className="landing-page"
>
  <h1
    className="ui header"
    data-bdd-selector="landingHeader"
  >
    React Redux Webpack Sample Application
  </h1>
  <div
    className="ui segment"
  >
    <h2
      className="ui header"
      data-bdd-selector="subHeader"
    >
      Sample UI Components
    </h2>
    <button
      className="ui primary button"
      data-bdd-selector="primaryButton"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Primary Button
    </button>
    <button
      className="ui secondary button"
      data-bdd-selector="secondaryButton"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Secondary Button
    </button>
  </div>
  <h2
    className="ui header"
    data-bdd-selector="routeHeader"
  >
    Read me and Examples
  </h2>
  <div
    className="ui list"
    role="list"
  >
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test1
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test2
    </div>
  </div>
</div>
`;
